### PR TITLE
Select the output language of the closure compiler

### PR DESCRIPTION
**Pull request by rixner**
A few small housekeeping things from #889, like create the ```tmp``` directory if it doesn't exist and use the production build to test in the web browser.

Mainly, however, sets the output language of the closure compiler to ECMAScript 2015 by default, but lets you select other variants if you prefer, as discussed in #896.  Speak up if you believe the default should be ECMAScript 5.

If people don't like this approach, I'm happy to go back to the drawing board.

### CODE REVIEW

File: support/run/test_template.ejs [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <html>
 <head>
     <title></title>
-    <script src="skulpt.js" type="text/javascript"></script>
+    <script src="skulpt.min.js" type="text/javascript"></script>
```
Comment (albertjan) [inline support/run/test_template.ejs]: is this change on purpose? I thought we used the dev build for browser debugging?

Comment (rixner) [reply support/run/test_template.ejs]: I needed to test that changing the output language produces a distribution file that still functions in a web browser.  As I thought about it, I started to think that we should be testing on the "real" distribution file under normal circumstances, as well.  The "brun" command still uses the dev build, as that seems more appropriate.  But, what does it matter if all the tests pass on the dev build?  They need to pass on the real build...

Ultimately, I think this is debatable and depends on what one is trying to do at the time.  Perhaps there needs to be a switch to allow either somehow?  Or I can switch it back.

**Comment by rixner** [comment]
Alternatively, maybe we should start with defaulting to ECMAScript 5 to avoid disruption.  If everyone finds themselves then always building ECMAScript 2015 (I'd make that a build command - ```build-es15```), then we can update it?

**Comment by albertjan** [comment]
I think 2015 should be fine for the majority of the users. If they're deploying it to a production environment that requires ES5 and they're building it themselves, they're probably able to figure out how to set that flag.

**Comment by albertjan** [comment]
I do think we should push a es2015 and a es5 version to skulpt-dist, so we can still help those people that don't build skulpt themselves.

File: support/run/test_template.ejs (hunk 1 above)
Comment (albertjan) [reply support/run/test_template.ejs]: A future update to this might be detecting which (minified or non minified) is available and adding that to the template.

Comment (rixner) [reply support/run/test_template.ejs]: Easily done, so I did.

**Comment by rixner** [comment]
@bnmnetp I'm good.  I am just reacting to any changes anyone would like.

**Comment by rixner** [comment]
This has sat with no activity for a week, can we either merge it or explain the problem with it?  Thanks!

@albertjan are you asking me to figure out how to push 2 versions to skulpt-dist?  Do you have a suggestion on how to name them?

```skulpt.min.js```  (es2015)
```skulpt.es5.min.js```

Or something else?

**Comment by albertjan** [comment]
That sounds good to me. i’m not too fussed on the name. Let’s merge this asap. 👍


________________________________
